### PR TITLE
enable requesting of rgson as a content type

### PR DESCRIPTION
**Pull request by tommysullivan**

### CODE REVIEW

**Comment by tommysullivan** [comment]
@bob-bins @aashreya can u two review this? tests are passing

**Comment by aashreya** [comment]
Hey @tommysullivan I might need a little background on the work done. Can we go over this together, maybe tomorrow ?

**Comment by tommysullivan** [comment]
@bob-bins @aashreya this is ready to merge, just moving the api closer to an actual rest api. still need to handle error cases and make the whole thing async but that can come later.

File: src/restguru/getCompleteProjection.ts [hunk 1]
```diff
@@ -1,32 +1,27 @@
...
+        .valueOrDefault(rgson)
 
-const mapArrayToCompleteProjection = (arrayRepresentation:string[]):JSONArray => 
-    arrayRepresentation.map(
-        url => getCompleteProjection({url, contentType: "sdfsdf" }).value
-    )
+const mapArrayToCompleteProjection = (arrayRepresentation:RGSONArray):JSONArray => 
+    arrayRepresentation.map(url => theInternet({url, contentType: "http://rest.guru/rgson/completeProjection"}).value)
 
 const mapDictionaryToCompleteProject = (dictionary:RGSONDictionary):JSONObject =>
```
Comment (bob-bins) [inline src/restguru/getCompleteProjection.ts]: could rename this to `mapDictionaryToCompleteProjection`

File: src/restguru/restClient.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { theInternet } from "./theInternet";
+
+export interface RestClient {
+    request(requestDescription:RequestDescription):Promise<HttpResponse>
+}
+
+export interface HttpResponse {
```
Comment (bob-bins) [inline src/restguru/restClient.ts]: are we leaving this blank!

Comment (tommysullivan) [reply src/restguru/restClient.ts]: i think so, we only check that there is a response in our tests for now

**Pull request merged**